File: code_festival_2017_qualc/C.py
s = input()
# 両端から見て同じならカウントを進める，どっちかにxだったら追加,それ以外はだめ

left = 0
right = len(s) - 1

ans = 0
while left < right:
    if s[left] == s[right]:
        left += 1
        right -= 1
    else:
        if s[left] == "x":
            left += 1
            ans += 1
        elif s[right] == "x":
            right -= 1
            ans += 1
        else:
            ans = -1
            break

print(ans)

File: code_festival_2017_qualc/C.cpp
#include <iostream>
#include <string>

using namespace std;

int main(){
    string s;
    cin>>s;
    int left,right;
    left=0;
    right=s.size()-1;

    int ans = 0;
    while(right>left){
        if(  s[right] == s[left] ){
            right--;
            left++;
        }else{
            if(  s[right] == 'x' ){
                ans += 1;
                right--;
            }else if(  s[left] == 'x' ){
                ans += 1;
                left++;
            }else{
                ans = -1;
                break;
            }
        }
    }
    cout<<ans<<endl;
    return 0;
}
